src/planscape/forsys/generate_projects_for_a_single_scenario.R

```diff
@@ -71,6 +73,29 @@ generate_projects_for_a_single_scenario <- function(
       mutate({{geo_wkt_field}} := st_as_text(geometry))
   }
 
+  # TODO: Should we even set a patchmax_proj_size if the client doesn't
+  # specify one?  The code would otherwise run with no limit (and seems to work).
+  if (identical(max_area_per_project_in_km2, "")) {
```

Comment:
Wdyt of simplifying this?

Ptal at the comment in line 220.

The gist is:
1) add validation to forsys_request_params to ensure either max acreage or max cost is defined.
2) in views.py, pass the value if it's defined; if it's undefined, pass the maximum possible value (planning area acreage for max area, planning area treatment cost for max cost)
3) in this R script, directly set proj_target_field to stand_cost_field, proj_target_value to the max cost, and patchmax_proj_size to the max acreage.

Reply:
(and apologies for sending you back to the drawing board - trying to keep the codebase as as concise as possible, open to push back)

Reply:
No worries.  Code has been updated so that forsys_request_params now enforces a default for max_area.  That said, I stopped short of forcing a maximum value for max_area for now, adding a TODO() to do this more gracefully, as the value of max_cost can really play a huge role on max_area maximums.